source/portable/NetworkInterface/STM32Hxx/NetworkInterface.c

```diff
@@ -304,6 +333,85 @@ static BaseType_t xSTM32H_NetworkInterfaceInitialise( NetworkInterface_t * pxInt
             HAL_ETH_DescAssignMemory( &( xEthHandle ), uxIndex, pucBuffer, NULL );
         }
 
+        #if ( ipconfigUSE_MDNS == 1 )
+            {
+                /* Program the MDNS address. */
+                prvMACAddressConfig( &xEthHandle, xMACEntry, ( uint8_t * ) xMDNS_MacAdress.ucBytes );
+                xMACEntry += 8;
+            }
+        #endif
+        #if ( ( ipconfigUSE_MDNS == 1 ) && ( ipconfigUSE_IPv6 != 0 ) )
+            {
+                prvMACAddressConfig( &xEthHandle, xMACEntry, ( uint8_t * ) xMDNS_MACAdressIPv6.ucBytes );
+                xMACEntry += 8;
+            }
+        #endif
+        #if ( ipconfigUSE_LLMNR == 1 )
+            {
```

Comment:
Why are we going through both line 349 and line 356 code when ipconfigUSE_IPv6 != 0

Reply:
For IPv4 and IPv6 we have different MAC addresses.

Reply:
Yes that is why, we should have  IPv4 check in place for first #if ipconfigUSE_LLMNR